R/sleuth.R

```diff
@@ -284,16 +283,66 @@ sleuth_prep <- function(
     ret$obs_norm_filt <- dplyr::semi_join(obs_norm, filter_df, by = 'target_id')
     ret$tpm_sf <- tpm_sf
 
+    path <- kal_dirs[1]
+    kal_path <- get_kallisto_path(path)
+    target_id <- as.character(rhdf5::h5read(kal_path$path, "aux/ids"))
+    num_transcripts <- length(target_id)
+    ret$bs_quants <- list()
+
+    msg("calculating bootstrap quantiles")
+    ret$bs_quants <- lapply(1:length(kal_dirs), function(i) {
+      kal_path <- get_kallisto_path(kal_dirs[i])
+      num_bootstrap <- as.integer(rhdf5::h5read(kal_path$path, "aux/num_bootstrap"))
+      msg(paste0("reading sample: ",names(kal_dirs[i])))
+      read_bootstrap_quant(fname = kal_path$path, num_bootstrap)
+    })
+    names(ret$bs_quants) <- names(kal_dirs)
+
+    transformation_function <- function(x) log(x + 0.5)
+
+    msg("calculating bootstrap variance")
+    ret$bs_summary <- do.call(cbind, lapply(seq_along(ret$kal), function(i) {
+      path <- kal_dirs[i]
+      samp_name <- names(path)
+      kal_path <- get_kallisto_path(path)
+      num_bootstrap <- as.integer(rhdf5::h5read(kal_path$path, "aux/num_bootstrap"))
+      bs_var <- read_bootstrap_statistics(
+        fname = kal_path$path,
+        num_bootstrap = num_bootstrap,
+        est_count_sf = est_counts_sf[[i]],
+        num_transcripts = num_transcripts,
+        transform = transformation_function
+        )
+      dot(i)
+      bs_var
+    }))
+
+    ret$target_id <- target_id
+
+    bs_test_summary <- adf(ret$bs_summary)
+    bs_test_summary$target_id <- target_id
+    bs_test_summary <- bs_test_summary[order(bs_test_summary$target_id), ]
+    bs_test_summary <- data.frame(varMeans =
```

Comment:
given that you perform some operations on this, I think it makes sense to leave it as a matrix until the last possible moment. please re factor so that you can perform all operations on the matrix, then simply have a very basic command that turns it into a data frame and adds the correct columns (`target_id`)